router-tests/testenv/testenv.go

```diff
@@ -126,6 +126,7 @@ type Config struct {
 	PrometheusRegistry                 *prometheus.Registry
 	ShutdownDelay                      time.Duration
 	NoRetryClient                      bool
+	DatadogTelemetry                   bool
```

Comment:
Can we make it generic so we don't have to introduce a flag per propagation method?